### ISSUE TITLE
generic edge restrictions

### ISSUE THREAD

**Issue by behrisch** [eclipse/sumo#5876]
It should be possible to define generic parameters for edges and vehicle (types) to restrict the accessibility of an edge. Most obvious properties are height, width and mass but also binary flags like classes of dangerous goods should be representable. The idea is to have a parameter (key, value) pair for the edge and for the vehicle type and a list of keys to use for comparison in a file or on the command line for the router. The values for all vehicle type parameters which get checked need to be smaller or equal than the corrsponding edge parameter for the vehicle to pass. Binary properties use 0 and 1, minimum restrictions are represented using negative values.

For performance reasons the parameters are parsed in advance into a vector of doubles for every vtype and edge. Furthermore a new router base class with an extended isProhibited function needs to be invented.

**Comment by behrisch** [comment]
it works only for the duarouter but that is enough for now
